File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/createnew/dialogs/askforname/presenter/AskForNamePresenter.java
package ua.meugen.android.pdfcomposer.ui.activities.createnew.dialogs.askforname.presenter;

import java.util.List;

import ua.meugen.android.pdfcomposer.model.data.PageContent;
import ua.meugen.android.pdfcomposer.ui.activities.base.fragment.presenter.MvpPresenter;
import ua.meugen.android.pdfcomposer.ui.activities.createnew.dialogs.askforname.state.AskForNameState;


public interface AskForNamePresenter extends MvpPresenter<AskForNameState> {

    String getName();

    List<PageContent> getPages();

    void onNameChanged(String name);
}

File: app/src/main/java/ua/meugen/android/pdfcomposer/ui/activities/createnew/dialogs/askforname/presenter/AskForNamePresenter.kt
package ua.meugen.android.pdfcomposer.ui.activities.createnew.dialogs.askforname.presenter

import ua.meugen.android.pdfcomposer.model.data.PageContent
import ua.meugen.android.pdfcomposer.ui.activities.base.fragment.presenter.MvpPresenter
import ua.meugen.android.pdfcomposer.ui.activities.createnew.dialogs.askforname.state.AskForNameState


interface AskForNamePresenter : MvpPresenter<AskForNameState> {

    val name: String?

    val pages: List<PageContent>?

    fun onNameChanged(name: String)
}
